src/main/java/org/jitsi/videobridge/Endpoint.java

```diff
@@ -1300,24 +1300,50 @@ public void addChannel(ChannelShim channelShim)
      */
     public void removeChannel(ChannelShim channelShim)
     {
-        synchronized (channelShims)
+        if (channelShims.remove(channelShim))
         {
-            switch (channelShim.getMediaType())
+            if (channelShims.isEmpty())
+            {
+                expire();
+            }
+            else
             {
-                case AUDIO:
-                    acceptAudio = false;
-                    break;
-                case VIDEO:
-                    acceptVideo = false;
-                    break;
+                refreshMediaDirection();
             }
+        }
+    }
 
-            channelShims.remove(channelShim);
-            if (channelShims.isEmpty())
+    /**
+     * Refresh media direction based on direction of
+     * {@link ChannelShim}. An endpoint can accept
+     * media, if its {@link ChannelShim} has either
+     * 'sendrecv' or 'recvonly' media direction.
+     */
+    public void refreshMediaDirection()
```

Comment:
`refreshMediaDirection` seems like a confusing name for this method since what's being modified is whether or not we accept audio/video (not the directions themselves).  `refreshAcceptedMediaTypes` maybe?

Reply:
@bbaldino thanks for suggestion, I think `updateAcceptedMediaTypes` would be even better name.